src/Data/Bin/Ordering.agda

```diff
@@ -0,0 +1,790 @@
+                                                    even<even (<-trans x<y y<z)
+<-trans {2[1+ x ]} {1+[2 y ]} {2[1+ .y ]} (even<odd x<y)
+                                          (odd<even (inj₂ refl)) =  even<even x<y
+<-trans {2[1+ x ]} {1+[2 y ]} {1+[2 z ]} (even<odd x<y) (odd<odd y<z) =
+                                                        even<odd (<-trans x<y y<z)
+<-trans {1+[2 x ]} {2[1+ y ]} {2[1+ z ]} (odd<even (inj₁ x<y)) (even<even y<z) =
+                                                   odd<even (inj₁ (<-trans x<y y<z))
+<-trans {1+[2 x ]} {2[1+ .x ]} {2[1+ z ]} (odd<even (inj₂ refl))
+                                          (even<even x<z) =  odd<even (inj₁ x<z)
+<-trans {1+[2 x ]} {2[1+ y ]} {1+[2 z ]} (odd<even (inj₁ x<y)) (even<odd y<z) =
+                                                         odd<odd (<-trans x<y y<z)
+<-trans {1+[2 x ]} {2[1+ .x ]} {1+[2 z ]} (odd<even (inj₂ refl)) (even<odd x<z) =
+                                                                 odd<odd x<z
+<-trans {1+[2 x ]} {1+[2 y ]} {2[1+ z ]} (odd<odd x<y) (odd<even (inj₁ y<z)) =
+                                                  odd<even (inj₁ (<-trans x<y y<z))
+<-trans {1+[2 x ]} {1+[2 y ]} {2[1+ .y ]} (odd<odd x<y) (odd<even (inj₂ refl)) =
+                                                             odd<even (inj₁ x<y)
+<-trans {1+[2 x ]} {1+[2 y ]} {1+[2 z ]} (odd<odd x<y) (odd<odd y<z) =
+                                                       odd<odd (<-trans x<y y<z)
+
+<⇒≯ : _<_ ⇒ _≯_
+<⇒≯ x<y y<x =  <-irrefl refl (<-trans x<y y<x)
+
+>⇒≮ : _>_ ⇒ _≮_
+>⇒≮ y<x =  <⇒≯ y<x
+
+<⇒≤ : _<_ ⇒ _≤_
+<⇒≤ = inj₁
+
+------------------------------------------------------------------------------
+-- resolving inequality comparison
+------------------------------------------------------------------------------
+
+open Tri
+
+<-cmp :  ∀ (x y) → Tri (x < y) (x ≡ y) (x > y)
+<-cmp zero     zero      = tri≈ ≮0     refl  ≮0
+<-cmp zero     2[1+ _ ]  = tri< 0<even (λ()) ≮0
+<-cmp zero     1+[2 _ ]  = tri< 0<odd  (λ()) ≮0
+<-cmp 2[1+ _ ] zero      = tri> (λ())  (λ()) 0<even
+<-cmp 2[1+ x ] 2[1+ y ]  with <-cmp x y
+... | tri< x<y _   _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)  where lt = even<even x<y
+... | tri≈ _   x≡y _ =  tri≈ (<-irrefl eq) eq (<-irrefl (sym eq))
+  where
+  eq = cong 2[1+_] x≡y
+... | tri> _ _ x>y =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt : 2[1+ x ] > 2[1+ y ]
+  gt = even<even x>y
+
+<-cmp 2[1+ x ] 1+[2 y ]  with <-cmp x y
+... | tri< x<y _    _ =  tri< lt (<⇒≢ lt) (<⇒≯ lt)
+  where
+  lt = even<odd x<y
+
+... | tri≈ _   x≡y _ =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  1+2x≡1+2y = cong 1+[2_] x≡y;  1+2x<2[1+x] = 1+[2x]<2[1+x] x
+
+  gt : 1+[2 y ] < 2[1+ x ]
+  gt = subst (_< 2[1+ x ]) 1+2x≡1+2y (1+[2x]<2[1+x] x)
+
+... | tri> _ _ y<x =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt : 1+[2 y ] < 2[1+ x ]
+  gt = odd<even (inj₁ y<x)
+
+<-cmp 1+[2 _ ] zero =  tri> (>⇒≮ gt) (>⇒≢ gt) gt
+  where
+  gt = 0<odd
```

Comment:
I've added some guidance to the style guide that in the case of very short terms like these then they should go on the same line as the `where` in order to save space, i.e. `where gt = 0<odd`

Reply:
I use this style feature now.